google/ggrc-core
# GGRC-8518 Hide function for the attributes works incorrect

# Issue description

Hide button should be shown for custom roles. 
Hide function works incorrect: when click Hide on an attribute, expected that only the attribute should be hidden, but actual all attribute are hidden

# Steps to test the changes

Case 1:
1. Open e.g Create  modal for any Object.
2. Fill all mandatory fields.
3. Click Hide icon on any GCA.
4. Click Save & Close button.
5. Open Create new issue modal again.
Expected: All GCA are displayed except one that was hidden in the step 3.

Case 2:
1. Open any Audit.
2. Click Create Template button on Audit Summary tab.
3. Click Create new assessment template link.
4. Set all mandatory fields.
5. Click Hide icon on the "Default Verifiers" field.
6. Click Save & Close button.
7. Click Create template button on Audit Summary tab one more time.
8. Click Create new assessment template link.
Expected: Hidden field should not be displayed.

Case 3:
1. Open Create modal for any Object.
2. Review Person type fields.
Expected: Hide icons are displayed for the all person type fields (except mandatory).

# Solution description

Added `hide` for custom roles.
Updated hide mechanism to hide attributes based on the title name(set by data attribute) instead on `tabindex`.

# Sanity checklist




# PR Review checklist
